### ISSUE TITLE
Virtualization of DetailsList not working inside Panel

### ISSUE THREAD

**Issue by joerneu** [OfficeDev/office-ui-fabric-react#6657]
### Please provide a reproduction of the bug in a codepen:

[Codepen sample](https://codepen.io/anon/pen/jemjMe)

### Bug Report 
 
- __Package version(s)__: 6.83.0
- __Browser and OS versions__: All
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: Normal
 
#### Describe the issue:
 
Virtualization of the DetailsList is not working.
 
#### Actual behavior:
 
I scroll down and the DetailsList displays an empty virtualized page.
 
#### Expected behavior:
 
Should always work.

#### To reproduce:

The Codepen sample displays the same DetailsList twice. Once in the main content area and another time in the Panel.

1. Scroll down in the main window -> works
2. Open the panel
3. Scroll down -> empty virtualized page

Check in Developer Tools:

The DOM for the DetailsList in the main content:

![maincontentdetailslist](https://user-images.githubusercontent.com/31096117/46795837-6bd28480-cd4b-11e8-8151-b56ec9da214d.jpg)

You can see that the Viewport's parent has Scroll Event Listeners attached.

The DOM for the DetailsList in the Panel's container:
 
![panelcontainerdetailslist](https://user-images.githubusercontent.com/31096117/46795876-8573cc00-cd4b-11e8-9c7d-350152fa779c.jpg)

The Scroll Event Listeners are missing.

P.S.: I disable scrolling of the Panel content so that I can use a Flex layout within the Panel. (Same for the main content.)

**Comment by micahgodbolt** [comment]
@kkjeer you've got this on your plate already right?

**Comment by eveleens** [comment]
Why do you need `display: flex` here? Is there more to the content? (Removing them gets rid of the issue.)

I've run into DetailsList virtualization issues before, and it's always when you're manipulating the layout - since virtualization is calculating based on layout.

**Comment by micahgodbolt** [comment]
Display flex is to allow the footer to be aligned at the bottom while the content can scroll. The virtualization bug is due to redundant overflow containers. I believe it's currently being resolved by @kkjeer

**Comment by micahgodbolt** [comment]
Can you one component be a component that renders a panel with stuff in the 'content' and the 'footer'? Or are you trying to abstract your component into something that COULD be placed into a panel?

**Comment by joerneu** [comment]
In all my cases the component is placed into a Panel. Theoretically it could be displayed anywhere, because there is nothing Panel-specific in the component. (Maybe the placement of the buttons.)

For example, look at the comment input here on Github. With "Write", "Preview", (scrolling) Text-Field, Text-Formatting commands, and at the bottom "Close and comment" and "Comment" buttons. The whole component could be placed in a Panel, a Dialog or inline on a page.

Does this anwser your question?

Maybe the Panel could have different components that depend on each other and extend the features, like a simple "blank" raw surface panel -> then with Close button and Header -> with Content area and Footer. A bit like List -> DetailsList -> GroupedList.

**Comment by micahgodbolt** [comment]
@joerneu yeah, I had a feeling you'd probably say that, but I thought I'd at least ask. So you want to be able to package up your own UI+buttons into a single component to drop into the panel. Once this code (#6680) is released, we should be in a good spot to easily modify the panel to fit your scenario.

**Comment by micahgodbolt** [comment]
actually, one last idea as you're talking about making a "blank" panel, we could easily make all of the 'contentInner' replaceable with an onRenderContent. This would leave you with a navigation and your content inside of a simple overflowY: auto flex container.

**Comment by KevinTCoughlin** [comment]
Let's keep this open @uifabric-bot

**Comment by joerneu** [comment]
@micahgodbolt : Yes, being able to replace "contentInner" is a good idea. This would enable getting rid of the CSS-"Hacks" from my Codepen sample.

I believe there was another issue where it was discussed if the Panel Header or Footer should scroll as well. My answer would be "It depends" and having a blank Panel would give the choice to the developer if required. Perfect!

(Sorry for the late reply - I missed the Github notification.)

**Comment by joerneu** [comment]
Did not mean to close the issue.

**Comment by micahgodbolt** [comment]
Digging back into this, to recap: In the non modal example scroll events are added above the ms-Viewport

![image](https://user-images.githubusercontent.com/1434956/50526987-475c8d80-0a9a-11e9-97a2-63ba70f4e723.png)


but in the panel, no such event is added 

![image](https://user-images.githubusercontent.com/1434956/50527006-593e3080-0a9a-11e9-8211-4a843a5d15be.png)

**Comment by KevinTCoughlin** [comment]
> 
> 
> Digging back into this, to recap: In the non modal example scroll events are added above the ms-Viewport
> 
> ![image](https://user-images.githubusercontent.com/1434956/50526987-475c8d80-0a9a-11e9-97a2-63ba70f4e723.png)
> 
> but in the panel, no such event is added
> 
> ![image](https://user-images.githubusercontent.com/1434956/50527006-593e3080-0a9a-11e9-8211-4a843a5d15be.png)

Thanks for the above @micahgodbolt, this is helpful debug information. We need to investigate further, but if the `scroll` events aren't being assigned then `findScrollableParent` must be returning `undefined` in `List.tsx` (my guess).

https://github.com/OfficeDev/office-ui-fabric-react/blob/bc132408840fe0f8de86e6f9c20bd3e9344657db/packages/office-ui-fabric-react/src/components/List/List.tsx#L296

**Comment by raveenone** [comment]
This issue is occurring on the All test plans page for AzureDevOps, any information around when this would get fixed or any workarounds would be greatly appreciated.

One thing we noticed was that this issue does not happen only on chrome but only on firefox and edge.

![image](https://user-images.githubusercontent.com/43638318/50955221-8e0ca780-14de-11e9-8ade-2f2744eb527f.png)

**Comment by micahgodbolt** [comment]
@raveenone are those panels where you have the details list, or some other container? We're trying to determine why this control is working in isolation, but fails under given scenarios.

**Comment by KevinTCoughlin** [comment]
> One thing we noticed was that this issue does not happen for us on chrome but only on firefox and edge.

@raveenone FWIW your Codepen repros the bug across Firefox, Edge, and Chrome.

**Comment by KevinTCoughlin** [comment]
@raveenone the root cause is that you are overriding `overflow-y` on the scrollable container which `DetailsList` uses for its on `'scroll'` event hook to invoke its virtualization logic as you mention in the original report:

>P.S.: I disable scrolling of the Panel content so that I can use a Flex layout within the Panel. (Same for the main content.)

See my _working_ modified version of your Codepen here: https://codepen.io/kevintcoughlin/pen/bOQpbM.

To get this to work, I've removed the following style override:

```diff
 /* Do not scroll panel */
 .ms-Panel-scrollableContent {
   display: flex;
-  overflow-y: hidden !important;
 }
```

The `DetailsList` looks for the nearest parent element which has `data-is-scrollable="true"`. In this case, that is `.ms-Panel-scrollableContent`. You are disabling the `overflow-y` behavior on that element, hence why virtualization on scroll is not working as expected.

You'll need to achieve your desired style(s) in a different way or modify which element is scrollable containing the `DetailsList` for virtualization to work.

**Comment by joerneu** [comment]
Ok, perfect, we are getting closer to a solution.

I already had a container for the list in my sample. Instead of changing the CSS as @KevinTCoughlin mentions and making the panel scrollable, I can keep the flex layout and fix the scrolling by enabling `autoflowY` on the list's container:

```diff
const listStyle = mergeStyles({
    flex: "1 1 auto",
    overflowX: "hidden",
>    overflowY: "auto"
});
```

**and** setting `data-is-scrollable` on the list's container:

`<div className={listStyle} data-is-scrollable="true">`

[Codepen](https://codepen.io/anon/pen/yGQOQy)

@KevinTCoughlin: Is the requirement of `data-is-scrollable` on the parent container documented somewhere? Should it be? Wouldn't it make more sense to add a ScrollableContainer or ListContainer component to Fabric React if there are specific requirements for the List's parent?

**Comment by KevinTCoughlin** [comment]
> @KevinTCoughlin: Is the requirement of `data-is-scrollable` on the parent container documented somewhere? Should it be? Wouldn't it make more sense to add a ScrollableContainer or ListContainer component to Fabric React if there are specific requirements for the List's parent?

Yes, its currently documented on https://developer.microsoft.com/en-us/fabric#/components/detailslist.

>My scrollable content isn't updating on scroll! What should I do?
>
>Add the data-is-scrollable="true" attribute to your scrollable element containing the DetailsList.
>
>By default, the List used within DetailsList will use the BODY element as the scrollable element. If you contain the List within a scrollable DIV using overflow: auto or scroll, the List needs to listen for scroll events on that element instead. On initialization, the List will traverse up the DOM looking for the first element with the data-is-scrollable attribute to know when element to listen to for knowing when to re-evaulate the visible window.

Though you aren't the first to miss it. We're open to suggestions on how to improve that documentation.

I know @ThomasMichon has opinions regarding this point and the component suggestion you raised. I thought this was one case which [ScrollablePane](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx) aims to solve.

**Comment by joerneu** [comment]
I looked at ScrollablePane and it seems that it is only used for Sticky components:

> ScrollablePane is a helper component designed to use in conjunction with a Sticky component. 

The documentation regarding scrolling was added after I had raised this issue. That's why I have not been aware of it.

In general, regarding usage and documentation: When I started with DetailsList I expected that it would already render the container for virtualized scrolling. From the documentation and the sample it is not clear that it is actually a long list with virtualized drawing that uses the parent container to decide what and when to draw itself. With the added documentation regarding scrolling it is clearer now. It could help if it was mentioned explicitly or if it had a sample. (Actually the available sample does use virtualization within a container but the parent for scrolling is outside of the sample's source and it is not displayed when I click on "Show code".)

---

BTW, the issue by @raveenone above might or might not be the same because he wrote that Chrome was not affected in his case. Should I close this issue?
